src/DurableTask.Netherite/OrchestrationService/NetheriteOrchestrationService.cs

```diff
@@ -88,58 +103,74 @@ public override string ToString()
 #else
             string configuration = "Release";
 #endif
-            return $"NetheriteOrchestrationService on {this.configuredTransport}Transport and {this.configuredStorage}Storage, {configuration} build";
+            return $"NetheriteOrchestrationService on {this.Settings.TransportChoice}Transport and {this.Settings.StorageChoice}Storage, {configuration} build";
         }
 
         /// <summary>
         /// Creates a new instance of the OrchestrationService with default settings
         /// </summary>
-        public NetheriteOrchestrationService(NetheriteOrchestrationServiceSettings settings, ILoggerFactory loggerFactory)
+        public NetheriteOrchestrationService(NetheriteOrchestrationServiceSettings settings, ILoggerFactory loggerFactory, IServiceProvider serviceProvider = null)
```

Comment:
It is preferable to take in the exact dependencies you need (to follow IoC / DI) vs `IServiceProvider`.

As with the other ctor comment, I recommend avoiding a breaking change by adding a separate ctor which takes in the new params.

Reply:
I don't prefer using a list of dependencies. It means we have to add/modify constructor overloads every time the list changes which is really inconvenient. Also, this does not work here since not all of the types are public.

How is this a breaking change?  the argument is optional.

Reply:
As mentioned in the other comment, it is breaking because code compiled against the previous dll will fail to find this constructor if the new version is dropped in place.

Reply:
Ok, I will create an overload then.